### PR TITLE
feat(Codebytes): move codebytes parent disc 351

### PR DESCRIPTION
**Pull request by BandanaKM**
### Overview

Please Note: this is the foundation for migrating the Codebytes package to a new repository. To our best of our knowledge, this repository is client-modules. I've confirmed with @Codecademy/web-platform that it's ok to merge packages that are in progress to main, as long as there are no consumers yet. 

We'll be merging this upon approval. 😄 

This first pull request does a couple things:

1. Moves the Codebytes parent component to client modules (`packages/codebytes/src/index.tsx`)
      - the Codebytes parent component currently just accepts text from whatever is calling it 
        
2. Renders a dummy Editor component (`packages/codebytes/src/editor.tsx`)
     - the Editor displays a text area
     - the onChange in text area sets the text whenever it is entered
     
3. Sets up storybook for client-modules: (`packages/styleguide/`)
    - (`packages/styleguide/.storybook`)
    - (`packages/styleguide/stories`)

4. Migrates a few essential package-related files that we have in gamut for the codebytes package, including `babel.config.js`, `tsconfig.json`, `package.json`. These are configuration files that are local to a package, and need not be in the root configuration. 

Screen Shot 2021-12-07 at 2 05 52 PM
 

### PR Checklist

- [x] Related to JIRA ticket: [DISC-351](https://codecademy.atlassian.net/browse/DISC-351)
- [x] I have run this code to verify it works

Note, I will make a pr with tests against the full integration branch once the UI is complete. 

### To test

- [x] Can be viewed on storybook locally by running `yarn start` from the root directory and viewing `localhost:6006`
- [x] Enter text, textarea will be updated. No errors should show in storybook. 

**IMPORTANT:** If your PR contains breaking changes, please remember to follow the instructions for breaking changes!

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/codebytes@40.4.2-alpha.4923e8.0<br/></pre>

**Comment by BandanaKM** [comment]
@JoshuaKGoldberg, i'm tagging you right now as an early reviewer with two questions, since you have context on this!

1. yarn format:verify/prettier is failing for me, but i'm not quite sure why. 
2. after creating a local babel.config.js, my yarn.lock file is much longer. does this generally seem ok?

**Comment by JoshuaKGoldberg** [comment]
> yarn format:verify/prettier

Try `yarn format`? See it in `package.json`, it passes `--write` and so should fix up any issues.

If that's not it, ugh, it might be something with endlines/whitespace?

> creating a local babel.config.js, my yarn.lock file is much longer

I bet those two are unrelated.

**Comment by BandanaKM** [comment]
> I bet those two are unrelated.

Actually, scratch that, it's the same length! @JoshuaKGoldberg

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/codebytes@40.4.2-alpha.5675d6.0<br/></pre>

File: packages/codebytes/babel.config.js
Comment (BandanaKM) [inline packages/codebytes/babel.config.js:14]: this file is copied directly from gamut for local packages. it also ensures that the compiler recognizes typescript

File: packages/codebytes/dist/index.js
Comment (BandanaKM) [inline packages/codebytes/dist/index.js:30]: this dist file is autogenerated

File: packages/codebytes/package.json
Comment (BandanaKM) [inline packages/codebytes/package.json:58]: using a copy of the package.json that we use for packages in gamut

File: packages/codebytes/src/editor.tsx
Comment (BandanaKM) [inline packages/codebytes/src/editor.tsx:9]: returning a textarea for now. we'll migrate monaco in a later ticket

File: packages/codebytes/src/stories/Codebytes.stories.tsx
Comment (BandanaKM) [inline packages/codebytes/src/stories/Codebytes.stories.tsx:16]: creating a story for Codebytes such that we can view the ui

File: packages/codebytes/tsconfig.json
Comment (BandanaKM) [inline packages/codebytes/tsconfig.json:7]: copied this over from gamut

**Comment by borisonr** [comment]
looks great @BandanaKM . i wonder if it would make sense to just merge this after approval instead of keeping the pr open until the rfc is done. we can always copy it into the new repo later, but it might be easier to parallel work and make sure things don't get stale if don't leave this pr open?

i also wonder if it might make sense to keep the naming conventions and file structure from static sites for now to make it clear what has been moved and what hasn't?

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/codebytes@40.4.2-alpha.afbdbd.0<br/></pre>

**Comment by BandanaKM** [comment]
> looks great @BandanaKM . i wonder if it would make sense to just merge this after approval instead of keeping the pr open until the rfc is done. we can always copy it into the new repo later, but it might be easier to parallel work and make sure things don't get stale if don't leave this pr open?

@borisonr The one question I had about merging once approved is that, if this repo is opensource, is it okay to have packages that are still a work in progress? I can bring it up with the web platform team too! 

> i also wonder if it might make sense to keep the naming conventions and file structure from static sites for now to make it clear what has been moved and what hasn't?

Yes, indeed! I am actually keeping the component file structure and names identical. There is logic in Codebytes parent that we do not need here, thus I didn't move those. The Editor component will be moved in full in the next pr!

**Comment by borisonr** [comment]
> > looks great @BandanaKM . i wonder if it would make sense to just merge this after approval instead of keeping the pr open until the rfc is done. we can always copy it into the new repo later, but it might be easier to parallel work and make sure things don't get stale if don't leave this pr open?
> 
> @borisonr The one question I had about merging once approved is that, if this repo is opensource, is it okay to have packages that are still a work in progress? I can bring it up with the web platform team too!
> 
> > i also wonder if it might make sense to keep the naming conventions and file structure from static sites for now to make it clear what has been moved and what hasn't?
> 
> Yes, indeed! I am actually keeping the component file structure and names identical. There is logic in Codebytes parent that we do not need here, thus I didn't move those. The Editor component will be moved in full in the next pr!

cool, imo it's fine to merge, but what do you think @Codecademy/web-platform?
and i think static sites uses slightly different file names? like CodeByteEditor instead of Codebytes. not a huge deal

**Comment by BandanaKM** [comment]
cc @JoshuaKGoldberg, do you think it's ok to merge a package that is in progress?

**Comment by BandanaKM** [comment]
> cc @JoshuaKGoldberg, do you think it's ok to merge a package that is in progress?

Confirming this is a yes as discussed on slack with jake and josh!

**Comment by j10sanders** [comment]
Maybe it would be useful to leave a comment in the storybook file that let's people looking at storybook that this is preliminary and not ready to be used?

**Comment by j10sanders** [comment]
Is there a reason to wait on the Monaco editor?  Seems it could be added to this PR?

**Comment by BandanaKM** [comment]
> Is there a reason to wait on the Monaco editor? Seems it could be added to this PR?

@j10sanders, the monaco editor is a bit more complex than it seems. the editor component itself has a lot of configuration (will be next), and that is the parent of the monaco editor. the monaco editor will need to be imported from gamut, which we didn't use in the original codebytes and will require some new dependencies. 

there's a bit more groundwork and some questions for reach which is why i am keeping it as a separate ticket!

**Comment by BandanaKM** [comment]
> Maybe it would be useful to leave a comment in the storybook file that let's people looking at storybook that this is preliminary and not ready to be used?

makes absolute sense!

**Review by j10sanders — APPROVED** [review]
Makes sense and looks good to me but I think needs Web-plat signoff

File: packages/codebytes/tsconfig.json
Comment (aionate0812) [inline packages/codebytes/tsconfig.json:4]: do we need a ` "rootDir": "src/"` property here? @JoshuaKGoldberg

File: packages/codebytes/package.json
Comment (aionate0812) [inline packages/codebytes/package.json:4]: probably want to reset to v0.0.1?

File: packages/codebytes/tsconfig.json
Comment (BandanaKM) [reply packages/codebytes/tsconfig.json:4]: thanks @aionate0812, good call out. i'll bring this up with josh when we pair!

File: packages/codebytes/package.json
Comment (BandanaKM) [reply packages/codebytes/package.json:4]: awesome, nice catch.

File: packages/codebyte-editor/.storybook/main.js
Comment (JoshuaKGoldberg) [inline packages/codebyte-editor/.storybook/main.js:2]: Let's stick to just one story extension. No need to allow multiple.

In general, let's try to use the same everything as Gamut.

File: packages/codebyte-editor/dist/editor.d.ts
Comment (JoshuaKGoldberg) [inline packages/codebyte-editor/dist/editor.d.ts:7]: These `dist` files should be gitignored.

File: packages/codebyte-editor/dist/stories/CodeByteEditor.stories.js
Comment (JoshuaKGoldberg) [inline packages/codebyte-editor/dist/stories/CodeByteEditor.stories.js:15]: Let's use the same story format as  Gamut, mdx is awesome and it'll be easier for us to unify tooling between repos.

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Great start!! We should pair soon but I don't want to block you waiting on me, so left some initial things.

File: packages/codebyte-editor/.storybook/main.js
Comment (JoshuaKGoldberg) [inline packages/codebyte-editor/.storybook/main.js:1]: Discussing in our pairing: we should have a standalone storybook package so there's only one, instead of stories per package.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/codebytes@0.0.2-alpha.093a46.0<br/>@codecademy/styleguide@52.2.3-alpha.093a46.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/codebytes@0.0.2-alpha.ae7edb.0<br/>@codecademy/styleguide@0.0.2-alpha.ae7edb.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/codebytes@0.0.2-alpha.e3edfc.0<br/>@codecademy/styleguide@0.0.2-alpha.e3edfc.0<br/></pre>

File: packages/codebytes/tsconfig.json
Comment (JoshuaKGoldberg) [reply packages/codebytes/tsconfig.json:4]: If it works without, you can remove it 😄 

`rootDir` is by default calculated as the longest common prefix for all included files. If that ends up being `src/` here then great.

https://aka.ms/tsconfig-reference

File: packages/styleguide/.storybook/main.ts
Comment (JoshuaKGoldberg) [inline packages/styleguide/.storybook/main.ts:41]:
```suggestion
      alias: emotionless(config.resolve.alias),
```

File: packages/styleguide/.storybook/preview-head.html
Comment (JoshuaKGoldberg) [inline packages/styleguide/.storybook/preview-head.html:7]: Not used, can be deleted?

File: packages/styleguide/jest.config.js
Comment (JoshuaKGoldberg) [inline packages/styleguide/jest.config.js:1]: Not used, can be deleted, I think?

File: packages/styleguide/typings.d.ts
Comment (JoshuaKGoldberg) [inline packages/styleguide/typings.d.ts:12]: Does anything break if you delete this file?

File: packages/styleguide/tsconfig.json
Comment (JoshuaKGoldberg) [inline packages/styleguide/tsconfig.json:21]:
```suggestion
```

Also the `typings` above

File: packages/styleguide/tsconfig.json
Comment (JoshuaKGoldberg) [inline packages/styleguide/tsconfig.json:11]: Hmm I want to say this actually isn't used?

```suggestion
```

File: packages/styleguide/tsconfig.json
Comment (JoshuaKGoldberg) [inline packages/styleguide/tsconfig.json:9]: I don't see these being used anywhere.

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Awesome so far! Some dead code to be deleted still.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/codebytes@0.0.2-alpha.518af8.0<br/>@codecademy/styleguide@0.0.2-alpha.518af8.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/codebytes@0.0.2-alpha.d83a50.0<br/>@codecademy/styleguide@0.0.2-alpha.d83a50.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/codebytes@0.0.2-alpha.c5fa0c.0<br/>@codecademy/styleguide@0.0.2-alpha.c5fa0c.0<br/></pre>

File: packages/codebytes/package.json
Comment (JoshuaKGoldberg) [reply packages/codebytes/package.json:11]: oop meant to post this last time :(

**Review by JoshuaKGoldberg — APPROVED** [review]
Looks great!! 👏 nicely done @BandanaKM!

File: packages/codebytes/package.json
Comment (BandanaKM) [reply packages/codebytes/package.json:11]: ah! ok, i just merged, I can include this in my next pull which i'm working on right now :)

**Pull request merged**
